ekncontent/ekncontent/eknc-dictionary-object-model.c

```diff
@@ -0,0 +1,202 @@
+/* Copyright 2017 Endless Mobile, Inc. */
+
+#include "eknc-dictionary-object-model.h"
+
+#include "eknc-utils-private.h"
+#include "eknc-content-object-model-private.h"
+
+/**
+ * SECTION:dictionary-object-model
+ * @title: dictionary Object Model
+ * @short_description: Access dictionary object metadata
+ *
+ * The model class for dictionary objects.
+ */
+typedef struct {
+  gchar *word;
+  gchar *definition;
+  gchar *partOfSpeech;
+} EkncDictionaryObjectModelPrivate;
+
+G_DEFINE_TYPE_WITH_PRIVATE (EkncDictionaryObjectModel,
+                            eknc_dictionary_object_model,
+                            EKNC_TYPE_CONTENT_OBJECT_MODEL)
+
+enum {
+  PROP_0,
+  PROP_WORD,
+  PROP_DEFINITION,
+  PROP_PART_OF_SPEECH,
+  NPROPS
+};
+
+static GParamSpec *eknc_dictionary_object_model_props [NPROPS] = { NULL, };
+
+static void
+eknc_dictionary_object_model_get_property (GObject    *object,
+                                           guint       prop_id,
+                                           GValue     *value,
+                                           GParamSpec *pspec)
+{
+  EkncDictionaryObjectModel *self = EKNC_DICTIONARY_OBJECT_MODEL (object);
+  EkncDictionaryObjectModelPrivate *priv = eknc_dictionary_object_model_get_instance_private (self);
+
+  switch (prop_id)
+    {
+    case PROP_WORD:
+      g_value_set_string (value, priv->word);
+      break;
+
+    case PROP_DEFINITION:
+      g_value_set_string (value, priv->definition);
+      break;
+
+    case PROP_PART_OF_SPEECH:
+      g_value_set_string (value, priv->partOfSpeech);
+      break;
+
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+    }
+}
+
+static void
+eknc_dictionary_object_model_set_property (GObject *object,
+                                           guint prop_id,
+                                           const GValue *value,
+                                           GParamSpec *pspec)
+{
+  EkncDictionaryObjectModel *self = EKNC_DICTIONARY_OBJECT_MODEL (object);
+  EkncDictionaryObjectModelPrivate *priv = eknc_dictionary_object_model_get_instance_private (self);
+
+  switch (prop_id)
+    {
+    case PROP_WORD:
+      g_clear_pointer (&priv->word, g_free);
```

Comment:
Since these properties are construct-only, you should not need to free the old value before setting the new one.